### ISSUE TITLE
"Downloader.app" will damage your computer. You should move it to the Bin.

### ISSUE THREAD

**Issue by TonnyCS** [vineetchoudhary/Downloader-for-Apple-Developers#33]
I got this error today, unfortunatelly I have no idea what to do. When I try to reinstall it via curl install.sh, the start fails with the error code 153.

```
The application cannot be opened for an unexpected reason, error=Error Domain=RBSRequestErrorDomain Code=5 "Launch failed." UserInfo={NSLocalizedFailureReason=Launch failed., NSUnderlyingError=0x600000d0a700 {Error Domain=NSPOSIXErrorDomain Code=153 "Unknown error: 153" UserInfo={NSLocalizedDescription=Launchd job spawn failed}}}
```

![Screenshot 2023-02-16 at 11 00 40](https://user-images.githubusercontent.com/71827410/219332979-c55e3b77-be18-46b6-b05a-6a660250d511.png)

**Comment by vineetchoudhary** [comment]
Hi, 

As per [this thread](https://developer.apple.com/forums/thread/663753), This issue is intermittent and happens 20% of time even if app codesigning and notarisation is clean.

Can you please confirm if you face this issue when you launch the app manually from Applications directory? If you are able to launch manually then I'll `open` command from [install script](https://github.com/vineetchoudhary/Downloader-for-Apple-Developers/blob/master/docs/install.sh).

**Comment by Moulick** [comment]
@vineetchoudhary did you make the release? I think I got a notification a couple of hours ago for the tag but an associated release does not exist, nor does a compiled app. Aria2c seems to have been added as an asset to the 2.2 release though https://github.com/vineetchoudhary/Downloader-for-Apple-Developers/releases/tag/2.2.0

**Comment by vineetchoudhary** [comment]
@Moulick, yes I released and removed 2.3.0 because I was still facing the same issue on another machine. I added `aria2c` to the asset because I'm planning to install `aria2c` separately. Because when `aria2c` is available in `Downloader.app` then macOS block the launch.

**Comment by kg-currenxie** [comment]
Same issue for me :/

**Comment by toineenzo** [comment]
Certificate seems expired?? I ones saw a video where somebody said a work around is to move the file around some directories and only after that, open it. But that doesn't seem to do the trick here. I think its just the expired certificate.
<img width="300" alt="SCR-20230224-2ns" src="https://user-images.githubusercontent.com/4921397/221065659-f458edbb-56dc-4765-ab59-6136946b85e9.png">

**Comment by Moulick** [comment]
Would love to know if there was any workaround to this problem. Even trying to self-sign via `codesign --force --deep -s - /Applications/Downloader.app` and codesign --force --deep --sign - /Applications/Downloader.app` does not work :(

**Comment by davidebeatrici** [comment]
Right click on the binary -> `Get Info` -> Check `Override Malware Protection`.

![](https://user-images.githubusercontent.com/5897523/230703329-d4204b7b-65fe-4384-8867-101911c680ff.png)

The application now crashes with `EXC_CRASH (SIGKILL (Code Signature Invalid))` though.

**Comment by Moulick** [comment]
Whoooo! I was actually able to make it work!

1. Download zip from https://github.com/vineetchoudhary/Downloader-for-Apple-Developers/releases/download/2.2.0/Downloader.app.zip
   1. unzip `Downloader.app.zip` 
   2. move the extracted `Downloader.app` into `/Applications` directory. 
   3. The link in readme has an expired certificate so will not work anymore
2. Try to launch once, MacOS will not allow it and give warnings about malware etc. 
    1. Dismiss the warnings. This is an important step else the below `Override Malware Protection` will not appear.
    2. Recent versions of MacOS (Sequoia onwards) will move the application to Trash immediately when opening for the first time. Move the application back from Trash to `/Applications`
3. Right-click the `Downloader.app`, Go to `Get Info`, and enable `Override Malware Protection`
4. Run `codesign --force --deep --sign - /Applications/Downloader.app`. This will sign the application with your own signature.
5. Run `codesign --force --deep --sign - /Applications/Downloader.app/Contents/Resources/aria2c`. This will sign the embedded aria2c binary.
6. Success!! Now you'll be able to launch and use `Downloader.app` without MacOS complaining.

**Comment by cbowns** [comment]
This is a lot of work to run an application that Apple revoked the certificate for.

**Comment by Moulick** [comment]
@cbowns true but till @vineetchoudhary is able to get a new cert and make a release signing both aria2c and his application,  this is the only workaround.

**Comment by Moulick** [comment]
Btw @vineetchoudhary, you don't have to release aria2c separately. I think as long as you sign it with your certificate and then include it in the `Downloader.app`, it should work. No need to even build the binary yourself, I think you can just get it from the official source during build time and sign it.
